lib/src/maintenance.dart

```diff
@@ -154,14 +173,64 @@ Future<Maintenance> detectMaintenance(
       await anyFileExists(analysisOptionsFiles, caseSensitive: true);
   final oldAnalysisOptions =
       analysisOptionsExists && !files.contains(currentAnalysisOptionsFileName);
+  var strongModeEnabled = false;
+  if (analysisOptionsExists) {
+    for (var name in analysisOptionsFiles) {
+      final file = new File(p.join(pkgDir, name));
+      if (await file.exists()) {
+        final content = await file.readAsString();
+        try {
+          final Map map = yaml.loadYaml(content);
+          final analyzer = map['analyzer'];
+          if (analyzer != null) {
+            strongModeEnabled = analyzer['strong-mode'] == true;
```

Comment:
Or! `analyzer['strong-mode']` is a Map – see https://www.dartlang.org/guides/language/analysis-options#specifying-strong-mode

Reply:
Good catch! Fixed...